pkg/secrethub/credentials/readers.go

```diff
@@ -0,0 +1,54 @@
+package credentials
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+)
+
+// Errors
+var (
+	ErrCannotFindHomeDir = errCredentials.Code("cannot_find_home_dir").ErrorPref(
+		"cannot find your home directory: %s",
+	)
+)
+
+type Reader interface {
+	Read() ([]byte, error)
```

Comment:
How is this different from the `io.Reader` interface? I get that `io.Reader` accepts a byte slice as an argument, but that's just implementation detail right?

Please add a clarifying comment as to why we have this interface here.

Reply:
I'll have a look again. The implementations might get a little complicated though.

Reply:
`readerFunc()` indeed becomes a little bit more complicated. But it works :-)